### ISSUE TITLE
combiner receives struct_pb2.ListValue instead of numpy list

### ISSUE THREAD

**Issue by Ridooo** [SeldonIO/seldon-core#686]
I am building a seldon deployment inference graph consists of a combiner which receives data from two child models. the models output a numpy list which is received by the combiner as google.protobuf.struct_pb2.ListValue. should it be received at the combiner the same type as it was sent from the model?

Model's output:
[[[211, 267, 320]], [[197, 251, 302]]]  >> numpy listcomp2 combiner input:

Received in combiner: 
[[values {number_value: 211.0}
 values { number_value: 267.0}
 values {  number_value: 320.0}]
[values {  number_value: 197.0}
 values {  number_value: 251.0}
 values {  number_value: 302.0}]]
 

seldon-core verion used in the combiner and model is '0.3.1'

**Comment by cliveseldon** [comment]
Yes that is strange. The code for the standard aggregate is

https://github.com/SeldonIO/seldon-core/blob/f0a940d628e2f8cdcc3fe6f7c516a86b100e51cd/python/seldon_core/seldon_methods.py#L214-L220

**Comment by Ridooo** [comment]
Here how to simulate the case:

```
from seldon_core.proto import prediction_pb2
from seldon_core.utils import array_to_grpc_datadef , extract_request_parts
from seldon_core.utils import array_to_list_value
import numpy as np

#output= [[[[311, 371, 430], [310, 372, 431]], [[214, 283, 348], [221, 290, 357]]],[[['EUS'], ['EUS']], [['EUS'], ['EUS']]]]
#output= [[[311, 371, 430], [310, 372, 431]], [[214, 283, 348], [221, 290, 357]]]
output1 = [[311],['hello']] 
output2 = [[311,350],['hello']]
output = np.array(output1)

meta = prediction_pb2.Meta()
names = []
default_data_type = "ndarray"
data = array_to_grpc_datadef(default_data_type, output, names)
seldon_msg = prediction_pb2.SeldonMessage(data=data, meta=meta)

features = extract_request_parts(seldon_msg) 
# here is what happens inside grpc_datadef_to_array
print(seldon_msg.WhichOneof("data_oneof"))
print(seldon_msg.data.WhichOneof("data_oneof"))
#np.array(np.array(msg.data.ndarray)[0][0][0])
np.array(seldon_msg.data.ndarray)
```

when you are using the output1,**features** will be:
array([['311'],
       ['hello']], dtype='<U5')
which is good, however, if you use output2, then **features** which what will be received by the combiner this input:

array([values {
  number_value: 311.0
}
values {
  number_value: 350.0
}
,
       values {
  string_value: "hello"
}
], dtype=object)

**Comment by Ridooo** [comment]
This issue can be fixed by using seldon_message_to_json() instead of np.array() in this function grpc_datadef_to_array for data_type == "ndarray"

**Comment by cliveseldon** [comment]
Can you test on latest 0.4.0 release?

**Comment by axsaucedo** [comment]
@Ridooo this is now fixed with #600 via #817.
